### PR TITLE
[react] Deprecate types related to classic components in favor of types from create-react-class

### PR DESCRIPTION
**Pull request by eps1lon**
The "classic component" notion is only relevant for `create-react-class`. Therefore, its types should live there instead of bloating types in `react`. This PR deprecates:
- `React.ClassicElement<Props>` (use `createReactClass.ComponentElement<Props, InstanceType<T>>`
- `React.ClassicComponentClass` (use `createReactClass.ClassicComponentClass`)
- `React.ClassicComponent` (use `createClass.ClassicComponent`)
- moves the deprecated `React.Mixin` and `React.ComponentSpec` into `create-react-class` that actually uses these types. Addons will use `createReactClass.ComponentSpec` going forward instead of the deprecated `React.ComponentSpec`

### CODE REVIEW

File: types/create-react-class/create-react-class-tests.ts [hunk 1]
```diff
@@ -75,32 +75,20 @@ const ClassicComponent: React.ClassicComponentClass<Props> = createReactClass<Pr
...
-const ClassicComponentNoState: React.ClassicComponentClass<{ text: string }> = createReactClass<{ text: string }>({
+const ClassicComponentNoState: createReactClass.ClassicComponentClass<{ text: string }> = createReactClass<
+    { text: string }
+>({
     render() {
         return DOM.div(this.props.text);
     },
 });
 
-// React.createFactory
```
Comment (eps1lon) [inline types/create-react-class/create-react-class-tests.ts:90]: I've moved all these tests to `react` and `react-dom`. They were actually testing that non-standard instance methods were preserved in these APIs which is unrelated to `create-react-class`.

**Comment by peterblazejewicz** [comment]
so those @deprecations were be left there for some time (you don't have any actual plan to remove/replace them, just want people to update source of the mixing, etc). Sorry to bother, I didn't write a line of React code recently (or at least a decent one)

**Comment by eps1lon** [comment]
Everything deprecated will be removed in 19 types. Added it to https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/64451

**Review by peterblazejewicz — APPROVED** [review]
@eps1lon 🙇🏻

**Comment by eps1lon** [comment]
Ready to merge
